src/process_utility.c

```diff
@@ -1571,10 +1571,10 @@ process_rename_column(ProcessUtilityArgs *args, Cache *hcache, Oid relid, Rename
 
 	dim = ts_hyperspace_get_dimension_by_name(ht->space, DIMENSION_TYPE_ANY, stmt->subname);
 
-	if (NULL == dim)
-		return;
-
-	ts_dimension_set_name(dim, stmt->newname);
+	if (dim)
+		ts_dimension_set_name(dim, stmt->newname);
+	if (ts_cm_functions->process_rename_cmd)
```

Comment:
Do we really this this new cross module function? We can use existing `tsl_ddl_command_start` hook instead

Reply:
I tried to refactor the process_utility code to directly call tsl_ddl_command_start without any additional tsl support functions.  It doesn't work as expected as some ddl functions are executed in multi transaction contexts. For this particular ddl command, the original hypertable has already been modified and the hypertable cache cannot find it by the time we get here with the original name.

Reply:
I share @pmwkaa's concern (and I have given similar feedback in the past). If the existing DDL forwarding doesn't work for some reason (I am not sure I understand why not), then we should at least use just one cross-module function for all DDL statements instead of adding a new one for each DDL. The reason the the hypertable cannot be found in the cache is because `CommandCounterIncrement()`. We have catalog update functions to deal with that without doing command-counter increment.

Reply:
@erimatnor This is the refactoring change we discussed. I spent a few days trying to refactor the code to get this to work: Setting the hypertable cache and passing it around didn't work as expected. I didn't want to spend more time fixing this at the cost of derailing this feature support. process_utility code needs a wider overhaul to work without adding new functions to the tsl code. We also have exactly the same issue calling out to sql functions that have implementations under tsl. This should also be addressed as part of the refactoring effort.